game_patch/multi/network.cpp

```diff
@@ -1049,10 +1186,80 @@ void __fastcall multi_io_stats_add_new(void *this_, int edx, int size, bool is_s
 
 FunHook<void __fastcall(void*, int, int, bool, int)> multi_io_stats_add_hook{0x0047CAC0, multi_io_stats_add_new};
 
+#pragma pack(push, 1)
+
+struct af_damage_notify_packet {
+    RF_GamePacketHeader header;
+    uint8_t player_id;
+    uint16_t damage;
+    uint8_t flags;
+};
+
+#pragma pack(pop)
+
+void af_process_damage_notify_packet(const void* const data, const size_t len, const rf::NetAddr& addr) {
+    // Receive: client <- server
+    if (!rf::is_multi || rf::is_server || rf::is_dedicated_server) {
+        return;
+    }
+
+    af_damage_notify_packet damage_notify_packet{};
+    if (len < sizeof(damage_notify_packet)) {
+        return;
+    }
+
+    std::memcpy(&damage_notify_packet, data, sizeof(damage_notify_packet));
+
+    const rf::Player* const player = rf::multi_find_player_by_id(damage_notify_packet.player_id);
+    if (!player) {
+        return;
+    }
+
+    const rf::Entity* const entity = rf::entity_from_handle(player->entity_handle);
```

Comment:
what's the point of finding the player/entity referenced by the packet? It is not used in any way. Does AF use it for something, e.g. for message?